File: app/src/main/java/com/popalay/cardme/business/splash/SplashInteractor.java
package com.popalay.cardme.business.splash;

import android.util.Log;

import com.popalay.cardme.data.repositories.settings.SettingsRepository;

import java.util.concurrent.TimeUnit;

import javax.inject.Inject;
import javax.inject.Singleton;

import io.reactivex.Completable;
import io.reactivex.schedulers.Schedulers;

@Singleton
public class SplashInteractor {

    private final SettingsRepository settingsRepository;

    @Inject SplashInteractor(SettingsRepository settingsRepository) {
        this.settingsRepository = settingsRepository;
    }

    public Completable start() {
        return Completable.mergeArray(Completable.timer(1, TimeUnit.SECONDS),
                settingsRepository.hasSettings()
                        .doOnSuccess(aBoolean -> Log.d("ss", "start: " + aBoolean))
                        .flatMapCompletable(hasSettings -> hasSettings ? Completable.complete()
                                : settingsRepository.save(settingsRepository.createDefault())))
                .subscribeOn(Schedulers.io());
    }
}

File: app/src/main/kotlin/com/popalay/cardme/business/splash/SplashInteractor.kt
package com.popalay.cardme.business.splash

import com.popalay.cardme.data.repositories.settings.SettingsRepository
import io.reactivex.Completable
import io.reactivex.schedulers.Schedulers
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class SplashInteractor @Inject internal constructor(private val settingsRepository: SettingsRepository) {

    fun start(): Completable = settingsRepository.hasSettings()
            .flatMapCompletable {
                if (it) Completable.complete()
                else settingsRepository.save(settingsRepository.createDefault())
            }
            .subscribeOn(Schedulers.io())
}
